File: backend/app/schemas/category.py
from pydantic import BaseModel, Field


class CategoryCreate(BaseModel):
    name_ar: str = Field(min_length=1, max_length=255)
    name_en: str | None = Field(default=None, max_length=255)
    icon: str | None = Field(default=None, max_length=255)
    sort_order: int = 0
    is_active: bool = True
    parent_id: str | None = None


class CategoryUpdate(BaseModel):
    name_ar: str | None = Field(default=None, min_length=1, max_length=255)
    name_en: str | None = Field(default=None, max_length=255)
    icon: str | None = Field(default=None, max_length=255)
    sort_order: int | None = None
    is_active: bool | None = None
    parent_id: str | None = None


class CategoryRead(BaseModel):
    id: str
    name_ar: str
    name_en: str | None
    icon: str | None
    sort_order: int
    is_active: bool
    parent_id: str | None = None  # new field — Flutter ignores unknown fields

    model_config = {
        "from_attributes": True,
    }

File: admin/src/types/category.ts
/**
 * Admin Categories module.
 * Mirrors `backend/app/schemas/category.py`. CategoryRead is shared with the
 * public list used by the Places module (`types/place.ts`).
 */
import type { CategoryRead } from './place'

export type { CategoryRead }

/** Body of `POST /api/v1/admin/categories` (`CategoryCreate`). */
export type CategoryCreatePayload = {
  name_ar: string
  name_en?: string | null
  icon?: string | null
  sort_order?: number
  is_active?: boolean
  parent_id?: string | null
}

/** Body of `PUT /api/v1/admin/categories/{id}` (`CategoryUpdate`). */
export type CategoryUpdatePayload = {
  name_ar?: string | null
  name_en?: string | null
  icon?: string | null
  sort_order?: number | null
  is_active?: boolean | null
  parent_id?: string | null
}